src/zenml/artifacts/external_artifact.py

```diff
@@ -99,7 +100,8 @@ def _validate_all(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         pipeline_name = values.get("pipeline_name", None)
         artifact_name = values.get("artifact_name", None)
         model_name = values.get("model_name", None)
-        model_version = values.get("model_version", None)
+        model_version_name = values.get("model_version_name", None)
+        model_version_number = values.get("model_version_number", None)
```

Comment:
I don't fully understand why this is needed. Feels like `model_version_name` is the model version and `model_version_number` is the version of the model version. If that's correct then I think this needs to be redesigned since have multi-layered versioning makes it to confusing in my opinion

Reply:
This supports what @stefannica  asked for.
Model Version is a container entity, which is represented by a display name (`model_version_name`) and continuous number (`model_version_number`). A display name is what the user can control, while a number is an auto-incremental integer assigned to it during creation. This change gives the user the flexibility to fetch the model version either by display name or by number.
These two are split into separate args, cause I thought it might be too confusing to understand why a "1" would be fetched by name, while 1 would be fetched by number while keeping it mixed in all internal methods.